test/integration/cronjob/cronjob_test.go

```diff
@@ -0,0 +1,205 @@
+
+	config := restclient.Config{Host: server.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("Error creating clientset: %v", err)
+	}
+	resyncPeriod := 12 * time.Hour
+	informerset := informers.NewSharedInformerFactory(clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "cronjob-informers")), resyncPeriod)
+	metrics.UnregisterMetricAndUntrackRateLimiterUsage("cronjob_controller")
+	cjc, err := cronjob.NewCronJobController(clientSet)
+	if err != nil {
+		t.Fatalf("Error creating CronJob controller: %v", err)
+	}
+	jc := job.NewJobController(informerset.Core().V1().Pods(), informerset.Batch().V1().Jobs(), clientSet)
+
+	return server, closeFn, cjc, jc, informerset, clientSet, config
+}
+
+func setupGarbageCollector(t *testing.T, config rest.Config, clientSet clientset.Interface, informerset informers.SharedInformerFactory, stopCh chan struct{}) {
+	client, err := dynamic.NewForConfig(&config)
+	if err != nil {
+		t.Fatal("Error creating dynamic client")
+	}
+	discoveryClient := cacheddiscovery.NewMemCacheClient(clientSet.Discovery())
+	restMapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
+	restMapper.Reset()
+	deletableResources := garbagecollector.GetDeletableResources(discoveryClient)
+	started := make(chan struct{})
+	close(started)
+	gc, err := garbagecollector.NewGarbageCollector(client,
+		restMapper,
+		deletableResources,
+		garbagecollector.DefaultIgnoredResources(),
+		informerset,
+		started,
+	)
+	syncPeriod := 5 * time.Second
+	go gc.Run(1, stopCh)
+	go gc.Sync(clientSet.Discovery(), syncPeriod, stopCh)
+}
+
+func newCronJob(name, namespace, schedule string) *batchv1beta1.CronJob {
+	zero64 := int64(0)
+	zero32 := int32(0)
+	return &batchv1beta1.CronJob{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CronJob",
+			APIVersion: "batch/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: batchv1beta1.CronJobSpec{
+			Schedule:                   schedule,
+			SuccessfulJobsHistoryLimit: &zero32,
+			JobTemplate: batchv1beta1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers:                    []corev1.Container{{Name: "foo", Image: "bar"}},
+							TerminationGracePeriodSeconds: &zero64,
+							RestartPolicy:                 "Never",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func cleanupCronJobs(t *testing.T, cjClient clientbatchv1beta1.CronJobInterface, name string) {
+	deletePropagation := metav1.DeletePropagationForeground
+	err := cjClient.Delete(name, &metav1.DeleteOptions{PropagationPolicy: &deletePropagation})
+	if err != nil {
+		t.Errorf("Failed to delete CronJob: %v", err)
+	}
+}
+
+func validateJobAndPod(t *testing.T, informerSet informers.SharedInformerFactory) {
+	if err := wait.Poll(10*time.Second, 120*time.Second, func() (bool, error) {
+		jobs := informerSet.Batch().V1().Jobs().Informer().GetIndexer().List()
```

Comment:
Since you are listing jobs, why not just using regular client and list them each time, any particular reason for informer here?